Source/Math/NcclComm.cpp

```diff
@@ -0,0 +1,123 @@
+//
+// Copyright (c) 2016, NVIDIA CORPORATION. All rights reserved.
+// Licensed under the MIT license. See LICENSE.md file in the project root for full license information.
+//
```

Comment:
Usually math should not contain any gpu related code. Only MathCuda should have it. I'm not sure what is your agreement with Mark though...